tests/behat/rate_limit.feature

```diff
@@ -23,152 +23,196 @@ Feature: Getting rate limited in different scenarios.
       | timelimit                    | 3600           |
       | browsersecurity              | securewindow   |
-      # A preflight check form (e.g. due to a set time limit or password) is required for rate limiting to take effect.
     And quiz "Test quiz name" contains the following questions:
       | question       | page |
       | First question | 1    |
+    # Start a quiz attempt as "student1".
     When I log in as "student1"
     And I am on "Course 1" course homepage
     And I follow "Test quiz name"
     And I press "Attempt quiz"
     And I press "Start attempt"
-    And I wait "1" seconds
-    And I switch to a second window
+    # New window should pop up.
+    Then I switch to a second window
     And I wait until the page is ready
-    And I check the quiz popup window depending on Moodle version
-    And I switch to a second window
+    # Attempt should start immediately for "student1".
+    Then I press "Start attempt"
+    And I should see "Answer the first question"
     And I close all opened windows
     And I log out
-    And I log in as "student2"
+    # Try to start a concurrent quiz attempt as "student2".
+    Then I log in as "student2"
     And I am on "Course 1" course homepage
     And I follow "Test quiz name"
     And I wait until the page is ready
     And I press "Attempt quiz"
     And I press "Start attempt"
-    And I should see "The server is processing many requests at the moment."
-    And I wait "2" seconds
+    # New window should pop up.
+    Then I switch to a second window
+    And I wait until the page is ready
+    # Rate limiting should kick in and prevent "student2" from starting the attempt.
+    Then I press "Start attempt"
+    And I should not see "Answer the first question"
+    And I should see "The server is processing many requests at the moment. Please wait until your quiz starts in a few seconds."
     And I click on ".close" "css" in the ".modal-content" "css_element"
-    And I click on ".btn-cancel" "css" in the ".mform" "css_element"
+    # TODO: The next 3 steps are a workaround because just using `And I press "Cancel"` causes an error in chromium -> `no such window: target window already closed`
+    And I switch to the main window
+    And I close all opened windows
+    And I reload the page
     And I press "Attempt quiz"
-    And I wait "5" seconds
     And I press "Start attempt"
-    And I wait "25" seconds
-    And I switch to a second window
-    Then I check the quiz popup window depending on Moodle version
+    # New window should pop up.
+    Then I switch to a second window
+    And I wait until the page is ready
+    # Rate limiting should still kick in and prevent "student2" from starting the attempt.
+    Then I press "Start attempt"
+    And I should not see "Answer the first question"
+    And I should see "The server is processing many requests at the moment. Please wait until your quiz starts in a few seconds."
```

Comment:
Ich bekomme auch mit firefox einen Fehler:

--- Failed steps:

```
001 Scenario: Cancelling secure window in a rate limited quiz with time limit                                                       # /var/www/html/mod/quiz/accessrule/ratelimit/tests/behat/rate_limit.feature:75
      And I should see "The server is processing many requests at the moment. Please wait until your quiz starts in a few seconds." # /var/www/html/mod/quiz/accessrule/ratelimit/tests/behat/rate_limit.feature:127
        "The server is processing many requests at the moment. Please wait until your quiz starts in a few seconds." text was not found in the page (Behat\Mink\Exception\ExpectationException)

```

Ich denke das liegt an den geänderten Sekondenzeiten beim warten, je nach dem wie schnell das System die Tests ausführt, bleiben dann weniger als 10 Sek beim Warten und das Popup "The server is processing many requests at the moment. Please wait until your quiz starts in a few seconds." wird dann nicht angezeigt.

Reply:
OK, aber das ist ja ein anderer Fehler. Wenn du `ms_between_attempts` wieder auf 25000 setzt, tritt dein Fehler nicht mehr auf? Dann können wir das auch wieder hochsetzen. Ich hatte nur damit rumgespielt und gehofft, dass wir das reduzieren können, weil ich die Dauer der Tests möglichst kurz halten wollte.

Aber konntest du den von mir erwähnten Fehler denn reproduzieren?

Reply:
Ich konnte den Fehler nicht reproduzieren, ich habe  ms_between_attempts wieder auf 25000 gesetzt und sowohl firefox als auch chrome sind erfolgreich durchgelaufen.